src/App.tsx

```diff
@@ -15,71 +16,96 @@ export const goods = [
-      Jam is selected
+export class App extends React.Component<{}, State> {
+  state = {
+    selectedProduct: 'Jam',
+  };
 
-      {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-      <button
-        data-cy="ClearButton"
-        type="button"
-        className="delete ml-3"
-      />
-    </h1>
+  render() {
+    return (
+      <main className="section container">
+        <h1 className="title is-flex is-align-items-center">
+          {
+            this.state.selectedProduct
+              ? `${this.state.selectedProduct} is selected`
+              : 'No goods selected'
+          }
 
-    <table className="table">
-      <tbody>
-        <tr data-cy="Good">
-          <td>
-            <button
-              data-cy="AddButton"
-              type="button"
-              className="button"
-            >
-              +
-            </button>
-          </td>
+          {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
+          {
+            this.state.selectedProduct && (
+              <button
+                data-cy="ClearButton"
+                type="button"
+                className="delete ml-3"
+                onClick={() => {
+                  this.setState({ selectedProduct: '' });
+                }}
+              >
+                {}
+              </button>
+            )
+          }
+        </h1>
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Dumplings
-          </td>
-        </tr>
+        <table className="table">
+          <tbody>
+            {
+              goods.map(product => (
+                <tr
+                  data-cy="Good"
+                  className={
+                    classNames({
+                      'has-background-success-light':
+                      this.state.selectedProduct === product,
+                    })
+                  }
+                >
+                  <td>
+                    {
+                      this.state.selectedProduct !== product
+                        ? (
+                          <button
+                            data-cy="AddButton"
+                            type="button"
+                            className="button"
+                            onClick={() => {
+                              this.setState({ selectedProduct: product });
+                            }}
+                          >
+                            +
+                          </button>
+                        )
+                        : (
+                          <button
+                            data-cy="RemoveButton"
+                            type="button"
+                            className="button is-info"
+                            onClick={() => {
+                              this.setState({ selectedProduct: '' });
+                            }}
+                          >
+                            -
+                          </button>
```

Comment:
you may try to unite these buttons into 1, and conditionally pass needed handlers, className, etc